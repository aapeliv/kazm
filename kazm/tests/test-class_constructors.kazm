class ScopeGreeter {
  ScopeGreeter() {
    println("Hi!");
  }

  ~ScopeGreeter() {
    println("Bye!");
  }
};

int main() {
  println("Top");
  {
    ScopeGreeter g;
    println("In scope");
  }
  println("Bottom");
}
