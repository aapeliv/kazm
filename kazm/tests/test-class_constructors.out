Top
Hi!
In scope
Bye!
Bottom
